abilities/music/commands.py

```diff
@@ -47,12 +50,59 @@ async def play(interaction: Interaction, song: str) -> None:
     else:
         voice_client = await channel.connect()
 
-    voice_client.play(
-        audio_source,
-        after=lambda e: print(f"Player error: {e}") if e else None,
-    )
+    if not voice_client.is_playing():
+        voice_client.play(
+            audio_source,
+            after=lambda e: print(f"Player error: {e}") if e else None,
+        )
+        await interaction.followup.send(f"Now playing: {song}")
+    else:
+        await interaction.followup.send(f"Added {song} to the queue.")
+        await queue.put((audio_source, song))
+
+    while voice_client.is_playing() or not queue.empty():
+        await asyncio.sleep(1)
+
+        if not voice_client.is_playing() and not queue.empty():
+            audio_source, song = await queue.get()
+            voice_client.play(
+                audio_source,
+                after=lambda e: print(f"Player error: {e}") if e else None,
+            )
+            await interaction.followup.send(f"Now playing: {song}")
```

Comment:
I think we should have a `await channel.disconnect()` after this while loop so that the bot leaves the channel once the queue is finished